The bypassed test program is confirmed to exit success on device.

--- a/configure.ac
+++ b/configure.ac
@@ -125,7 +125,7 @@ if test "$thread" != "no"; then
 	elif test "$OS" = "SOLARIS"; then
 	    LDFLAGS="$LDFLAGS -lpthread"
 	fi
-	AC_RUN_IFELSE([AC_LANG_SOURCE([[
+	AC_LINK_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 void init_routine() { return; }
 int main() {
